nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ConvertJSONtoCSV.java

```diff
@@ -0,0 +1,360 @@
+            session.transfer(flowFile, RELATIONSHIP_FAILURE);
+        }
+
+    }
+
+    private Set<String> getRemoveKeySet() {
+        Set<String> setRemove = new HashSet<>();
+
+        final String[] fields = removeFields.split(",");
+        for (String field : fields) {
+            setRemove.add(field);
+        }
+        return setRemove;
+    }
+
+    private static class JSONParser {
+
+        private static final Class<?> JSON_OBJECT = JSONObject.class;
+
+        private static final Class<?> JSON_ARRAY = JSONArray.class;
+
+        private static Set<String> removeKeySet = new HashSet<>();
+
+        public static List<Map<String, String>> parseJSON(String json, Set<String> removeKeys) throws JSONException {
+            List<Map<String, String>> flatJson = null;
+            removeKeySet = removeKeys;
+
+            // we don't know if we are dealing with a JSON object or an array
+            // not handling this properly causes un-necessary exception to be thrown
+            // instantiate to a generic object and check what we are dealing with
+            Object objJSON = new JSONTokener(json).nextValue();
+
+            try {
+                if (objJSON instanceof JSONObject) {
+                    JSONObject jsonObject = new JSONObject(json);
+                    flatJson = new ArrayList<Map<String, String>>();
+                    flatJson.add(parse(jsonObject));
+                } else if (objJSON instanceof JSONArray) {
+                    flatJson = handleAsArray(json);
+                }
+            } catch (JSONException ex) {
+                throw new JSONException("JSON might be malformed, please check and make sure it is valid JSON format: " + ex);
+            }
+
+            return flatJson;
+        }
+
+        public static Map<String, String> parse(JSONObject jsonObject) {
+            Map<String, String> flatJson = new LinkedHashMap<String, String>();
+            try {
+                denormalize(jsonObject, flatJson, "");
+            } catch (Exception ex) {
+                throw ex;
+            }
+            return flatJson;
+        }
+
+        public static List<Map<String, String>> parse(JSONArray jsonArray) {
+            JSONObject jsonObject = null;
+            List<Map<String, String>> flatJson = new ArrayList<Map<String, String>>();
+            int length = jsonArray.length();
+
+            for (int i = 0; i < length; i++) {
+                jsonObject = jsonArray.getJSONObject(i);
+                Map<String, String> stringMap = parse(jsonObject);
+                flatJson.add(stringMap);
+            }
+
+            return flatJson;
+        }
+
+        private static List<Map<String, String>> handleAsArray(String json) throws JSONException {
+            List<Map<String, String>> flatJson = null;
+
+            try {
+                JSONArray jsonArray = new JSONArray(json);
+                flatJson = parse(jsonArray);
```

Comment:
Can this be parallelized somehow? Fork/Join or whatever? Not sure about overhead, just asking :)